backend/frontend/coursecreation/createcourse.html

```diff
@@ -0,0 +1,210 @@
+                  "title": "Week {{idx}}",
+                  "htmlClass": "week",
+                  "properties": {
+                    "dateRangeInput": {
+                      "type": "object",
+                      "properties": {
+                        "start": {
+                          "type": "date",
+                          "description": "start date"
+                        },
+                        "end": {
+                          "type": "date",
+                          "description": "end date"
+                        }
+                      }
+                    },
+                    "assignments": {
+                      "type": "array",
+                      "title": "assignments",
+                      "htmlClass": "assignments",
+                      "uniqueItems": true,
+                      "items": {
+                        "type": "object",
+                        "properties": {
+                          "title": {
+                            "type": "string",
+                            "description": "title"
+                          },
+                          "markdown": {
+                            "type": "textarea",
+                            "description": "contents"
+                          }
+                        }
+                      }
+                    },
+                    "topics": {
+                      "type": "array",
+                      "title": "topics",
+                      "htmlClass": "topics",
+                      "minItems": 1,
+                      "uniqueItems": true,
+                      "items": {
+                        "type": "object",
+                        "properties": {
+                          "title": {
+                            "type": "string",
+                            "description": "title"
+                          },
+                          "resources": {
+                            "type": "array",
+                            "title": "resources",
+                            "htmlClass": "resources",
+                            "minItems": 1,
+                            "uniqueItems": true,
+                            "items": {
+                              "type": "object",
+                              "properties": {
+                                "title": {
+                                  "type": "string",
+                                  "description": "title"
+                                },
+                                "markdown": {
+                                  "type": "textarea",
+                                  "description": "contents"
+                                }
+                              }
+                            }
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        },
```

Comment:
Is there any way to load this in from the schema? This is a lot of retyping, and if we ever change the schema, we've got to make changes in two places, which is a pain.